### PR TITLE
Chore/Repo maintenance

### PR DESCRIPTION
**Pull request by iGxnon**
Please briefly answer these questions:

Base on #1 

This PR add some repository maintenance files

* what problem are you trying to solve? (or if there's no problem, what's the motivation for this change?)

* what changes does this pull request make?

* are there any non-obvious implications of these changes? (does it break compatibility with previous versions, etc)

### CODE REVIEW

**Review by themanforfree — APPROVED** [review]

**Comment by themanforfree** [comment]
Please add a pre commit config file.

File: deploy/src/config.rs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#[non_exhaustive]
+#[command(author, version, about, long_about = None)]
+pub struct Config {
+    /// The address on which the HTTP server will listen to
+    #[arg(long, default_value = "0.0.0.0:8080")]
+    pub listen_addr: String,
+    /// The namespace to deploy
+    #[arg(long, default_value = "default")]
+    pub namespace: String,
+    /// Enable operator to work in all namespaces
```
Comment (Phoenix500526) [inline deploy/src/config.rs:14]: How many namespaces do we have? In what namespaces do we need to enable the operator to work and in what not?

**Comment by bsbds** [comment]
It seems only the last commit is needed in this PR, please rebase.

File: deploy/src/crd.rs [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+        S3 { s3: S3Spec },
+        #[serde(rename = "pv")]
+        PV { pv: String },
+    }
+
+    /// Xline cluster backup S3 specification
+    #[derive(Serialize, Deserialize, Clone, Debug, JsonSchema)]
+    pub(crate) struct S3Spec {
+        /// S3 bucket path to use for backup
+        /// TODO validate
```
Comment (Phoenix500526) [inline deploy/src/crd.rs:90]: Can we validate it in this pr? Or we'd better to leave an issue to track it.

**Comment by Phoenix500526** [comment]
Why does this pr based on #1? It looks like we only need the last commit.

**Comment by iGxnon** [comment]
> Why does this pr based on #1? It looks like we only need the last commit.

Before #1, our code has not yet initialized the crate, so the CI scripts introduced by this PR will fail.

File: deploy/src/config.rs (hunk 1 above)
Comment (iGxnon) [reply deploy/src/config.rs:14]: These codes are introduced in #1.
The number of namespaces depends on the situation of k8s, and this parameter is determined by the user. By the way, there is a `cluster_wide` parameter below, which will allow deploy-operator to work in all namespaces.

**Comment by iGxnon** [comment]
> Please add a pre commit config file.

done

File: deploy/src/crd.rs (hunk 2 above)
Comment (iGxnon) [reply deploy/src/crd.rs:85]: I'll fix it in #8

File: README.md [hunk 3]
```diff
@@ -1,2 +1,4 @@
 # xline-operator
-The `xline-operator` is a powerful tool designed to automate the process of bootstrapping, monitoring, snapshotting, and recovering an xline cluster on Kubernetes.  
+
+The xline-operator is a powerful tool designed to automate the process of bootstrapping, monitoring, snapshotting, and
+recovering a xline cluster on Kubernetes.
```
Comment (Phoenix500526) [inline README.md:4]: It's "an" xline cluster, not a xline cluster.

Comment (iGxnon) [reply README.md]: done

**Review by Phoenix500526 — DISMISSED** [review]

**Comment by bsbds** [comment]
please add the typo check in the CI

**Comment by iGxnon** [comment]
> please add the typo check in the CI

done

**Review by bsbds — APPROVED** [review]

**Pull request merged**